### DISCUSSION TITLE
MPD service & pydpiper

### DISCUSSION THREAD

**Discussion by kosto-stavr** [rern/rAudio#89] [General]
Hi to everybody and a happy new year!
In rAudio the MPD service still uses the ”localhost:6600”, or something have change?
I’m asking because on my pydpiper script, which I use for the oled display, I getting :

2021-01-07 12:26:37,596:DEBUG:Connecting to MPD service on localhost:6600
2021-01-07 12:26:47,621:DEBUG:Could not connect to MPD 

BTW does anyone use successfully the pydpiper ?
thanks

**Comment by rern** [comment]
Please try: 
System > GPIO Devices > LCD - Character
(without pydpiper)

**Reply by kosto-stavr** [reply]
Hi rern,
Following your suggestion I tried the LCD-Character.
My display is the : “Winstar WEH001602A Ecran Oled 16x2 HD44780”
(https://www.audiophonics.fr/en/screens-vu-meters/winstar-weh001602a-ecran-oled-16x2-blanc-hd44780-p-8512.html)
My wiring is this:
|---OLED Display---|      |---Raspberry Pi---|
| Pin  Description |          | Pin  Description |
   1   Vss (GND)          <---->   6   GND
   2   VDD (+5.0V)      <---->   2   +5.0V
   3   nc
   4   RS         <------------>  26   GPIO 7 
   5   R/W      <------------>  20   GND
   6   E           <------------>  24   GPIO8
   7   DB0
   8   DB1
   9   DB2
  10   DB3
  11   DB4   <------------>  22   GPIO25
  12   DB5   <------------>  18   GPIO24
  13   DB6   <------------>  16   GPIO23
  14   DB7   <------------>  10   GPIO15

I use the following setup, but nothing happens.

<img src="https://user-images.githubusercontent.com/13005709/103993356-25746600-519e-11eb-92da-2f0866046127.PNG" width=300>
What I am doing wrong ?

**Comment by kosto-stavr** [comment]
OK
I found a working solution!
If in mpd.conf I change the default “bind_to_address   "0.0.0.0"” to “bind_to_address   "127.0.0.1"”  then the  pydPiper works OK as before.

Does this change is acceptable or I have to take care something else as well ?

Anyhow thanks a lot rern for all of your contribution.
Regards 
Stavros

**Comment by shinzuka** [comment]
I have exactly the same gpio configuration but it is impossible to set R/W to 20 not available in the list.

![Capture d’écran du 2024-05-31 10-59-28](https://github.com/rern/rAudio/assets/68748242/8cbccb0f-dda2-4f4e-a190-528bcf6d21fa)
